packages/perseus/src/widgets/passage-ref.jsx

```diff
@@ -125,7 +135,18 @@ class PassageRef extends React.Component<Props, State> {
 
         this._deferredUpdateRange();
 
-        this._throttledUpdateRange = _.throttle(this._deferredUpdateRange, 500);
+        function throttle(func, timeFrame) {
+            let lastTime: number | Date = 0;
+            return function (...args): void {
+                const now = new Date();
+                if (now - lastTime >= timeFrame) {
+                    func(...args);
+                    lastTime = now;
+                }
+            };
+        }
+
```

Comment:
Compared to [_.throttle](https://underscorejs.org/docs/modules/throttle.html), this function will miss the "trailing edge"; e.g. if you do two keystrokes close to one another and then nothing for a while, `_.throttle` will handle the first one, and after 500ms handle the second one. This function will handle the first one and drop the second one, which seems undesirable?
I would imagine we have a function like this defined somewhere in our codebase, have you checked around for one?

Reply:
Yeah, throttling is tricky. I'd avoid trying to declare your own and recommend sticking with underscore for now. Either way this would be the wrong place to declare a generic helper function. I wonder if something like Wonder-Blocks has utility helpers it exports?

Reply:
Thank you for this feedback! I will do some research to see if I can find another function in the codebase or if Wonder-Blocks has something we can use. Otherwise, I will just switch it back to using underscore for now.

Reply:
Just switched it back to the underscore library for now.